evalml/pipelines/components/transformers/preprocessing/text_featurizer.py

```diff
@@ -119,7 +119,7 @@ def transform(self, X, y=None):
         if X_nlp_primitives.isnull().any().any():
             X_nlp_primitives.fillna(0, inplace=True)
 
-        X_lsa = self._lsa.transform(X[self._text_columns]).to_dataframe()
+        X_lsa = self._lsa.transform(X_ww[self._text_columns]).to_dataframe()
```

Comment:
This was the first fix required for this bug. We need to pass the woodwork type info to the LSA component, otherwise it will run woodwork inference again from scratch and the newly reset types will be used by all the downstream components!